tools/code-owners-parser/CodeOwnersParser/CodeownersEntry.cs

```diff
@@ -106,11 +106,69 @@ public void ParseOwnersAndPath(string line)
             line = ParsePath(line);
             line = RemoveCommentIfAny(line);
 
-            foreach (string author in SplitLine(line, OwnerSeparator).ToList())
+            // If the line doesn't contain the OwnerSeparator AKA no owners, then the foreach loop below
+            // won't work. For example, the following line would end up causing "/sdk/communication" to
+            // be added as an owner when one is not listed
+            // /sdk/communication/
+            if (line.Contains(OwnerSeparator))
```

Comment:
I think it would be nice to have automated unit tests covering this new logic that run as part of the automated regression test suite, not manually.